raiden/network/pathfinding.py

```diff
@@ -305,6 +305,41 @@ def configure_pfs_or_exit(
     return pathfinding_service_info
 
 
+def check_pfs_for_production(
+    service_registry: Optional[ServiceRegistry], pfs_info: PFSInfo
+) -> None:
+    """ Checks that the PFS in `pfs_info` is registered in the service registry
+    and that the URL matches.
+
+    Should only be called in production mode.
+    """
+    if service_registry is None:
+        click.secho(
+            f"Cannot verify registration of PFS because no service registry is set"
+            f"Raiden will shut down. Please set the service registry."
+        )
+        sys.exit(1)
+
+    pfs_registered = service_registry.has_valid_registration(
+        block_identifier=service_registry.client.get_confirmed_blockhash(),
+        service_address=pfs_info.payment_address,
+    )
+    registered_pfs_url = service_registry.get_service_url(
+        block_identifier=service_registry.client.get_confirmed_blockhash(),
+        service_address=pfs_info.payment_address,
+    )
+    pfs_url_matches = registered_pfs_url == pfs_info.url
+    if not (pfs_registered and pfs_url_matches):
+        click.secho(
```

Comment:
Do we really want to have ui code here? Shouldn't we raise some kind of exception and let the ui code present the error?

Reply:
This is the pattern currently used in that code, but I agree we should remove it as it makes it harder to understand the flow and also irritates tools like `mypy`.

However, I think that should be done in a different PR.

Reply:
Created https://github.com/raiden-network/raiden/issues/4615